src/background/Wallet/Wallet.ts

```diff
@@ -1395,18 +1377,22 @@ class PublicController {
     if (!this.wallet.allowedOrigin(context, currentAddress)) {
       throw new OriginNotAllowed();
     }
-    const transaction = params[0];
+    const [transaction, { clientScope } = { clientScope: undefined }] = params;
     invariant(transaction, () => new InvalidParams());
     const isDeviceWallet = currentWallet && isDeviceAccount(currentWallet);
+    const searchParams = new URLSearchParams({
+      origin: context.origin,
+      transaction: JSON.stringify(transaction),
+    });
+    if (clientScope) {
+      searchParams.append('clientScope', clientScope);
```

Comment:
Not sure is it better, but can it be more obvious to create a meta param like `analytics_context` where it is possible to serialise any json and put as many params as needed in the future. This data will be separate from the rest which can be more clear (or not, not sure :))

Reply:
Not a bad idea, but I'm hesitant about such generalizations. I'd go for explicit for as long as we don't hit its limitations. It's hard to predict whether this case will ever evolve